packages/marshal/src/helpers/passStyle-helpers.js

```diff
@@ -75,14 +75,6 @@ export const checkNormalProperty = (
     return reject(X`${q(propertyName)} property expected: ${candidate}`);
   }
   return (
-    (nameType === undefined ||
-      // eslint-disable-next-line valid-typeof
-      typeof propertyName === nameType ||
-      reject(
-        X`${q(propertyName)} must be a ${q(
-          nameType,
-        )}-named property: ${candidate}`,
-      )) &&
```

Comment:
So was all this simply unneeded?

Reply:
Yes, because `propertyName` comes from a literal or number loop iteration variable at all but one call site, and making assertions about its type in those cases is pointless.